src/components/ArticleCard.jsx

```diff
@@ -1,26 +1,46 @@
 import { useEffect, useState } from "react";
 import { fetchArticleById } from "../utils/api";
-import { useParams } from "react-router-dom";
+import { useParams, Link } from "react-router-dom";
+import Comments from "./Comments";
 
-const ArticleCard = () => {
+export default function ArticleCard() {
   const [articleCard, setArticleCard] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
   const { article_id } = useParams();
 
   useEffect(() => {
-    fetchArticleById(article_id).then((response) => {
-      setArticleCard(response.data.article);
-    });
+    fetchArticleById(article_id)
+      .then((response) => {
+        setArticleCard(response.data.article);
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        setError(error);
+        setIsLoading(false);
+      });
   }, []);
 
+  if (isLoading)
+    return (
+      <div className="loading-container">
+        <p>{`Loading please wait...`}</p>
+      </div>
+    );
+
+  if (error) return <Error error={error} />;
+
   return (
     <>
       <h3>{articleCard.title}</h3>
       <p>Author: {articleCard.author}</p>
       <p>Votes: {articleCard.votes}</p>
       <img src={articleCard.article_img_url}></img>
       <p>{articleCard.body}</p>
-      <button>Comments</button>
+      <Link to={`/articles/${articleCard.article_id}/comments`}>
+        <button>View All Comments</button>
+      </Link>
+      <Comments />
```

Comment:
Is this a button to view all comments but also the comments are rendered underneath anyway? I'd probably just pick one or the other, you don't necessarily need a separate page for comments they can just live in the same page underneath...upto you!